generator/src/test/java/com/scottlogic/deg/generator/restrictions/FieldSpecFactoryTests.java

```diff
@@ -58,8 +59,8 @@ void construct_twoInstancesOfStringHasLengthConstraintCalledWithEqualValues_retu
         StringHasLengthConstraint firstConstraint = new StringHasLengthConstraint(testField,HelixStringLength.create(20));
         StringHasLengthConstraint secondConstraint = new StringHasLengthConstraint( testField, HelixStringLength.create(20));
 
-        final FieldSpec firstInstance = firstConstraint.toFieldSpec();
-        final FieldSpec secondInstance = secondConstraint.toFieldSpec();
+        final RestrictionsFieldSpec firstInstance = (RestrictionsFieldSpec) firstConstraint.toFieldSpec();
+        final RestrictionsFieldSpec secondInstance = (RestrictionsFieldSpec) secondConstraint.toFieldSpec();
 
         assertEquals(firstInstance.getRestrictions(), secondInstance.getRestrictions());
```

Comment:
Would it be worth adding more of these methods to FieldSpec to avoid all of these casts?

Reply:
no